packages/runtime/src/useCases/common/validation/GenericInputValidator.ts

```diff
@@ -0,0 +1,27 @@
+import { RuntimeErrors } from "../RuntimeErrors";
+import { JsonSchema } from "../SchemaRepository";
+import { SchemaValidator } from "./SchemaValidator";
+import { ValidationFailure } from "./ValidationFailure";
+import { ValidationResult } from "./ValidationResult";
+
+export class GenericInputValidator<T extends object> extends SchemaValidator<T> {
```

Comment:
I'm very confused, could you elaborate the naming here? Why is this InputValidator more GENERIC than the class that it extends?

Reply:
Also, Schema is missing in the name.

Reply:
This is not fully resolved.